src/PluginsSystem/Microsoft.Performance.Toolkit.Plugins.Runtime/PluginsSystem.cs

```diff
@@ -84,13 +92,15 @@ public sealed class PluginsSystem
         /// </returns>
         public static PluginsSystem CreateFileBasedPluginsSystem(
             string root,
+            Func<string, Task<bool>> loadPluginFromDirectory,
```

Comment:
we might want to consider, in the future, just making this an interface `IPluginDirectoryLoader`